src/Components/TruckCard/TruckCard.js

```diff
@@ -6,17 +6,76 @@ import notUpvoted from '../../assets/notUpvoted.png'
 import downvoted from '../../assets/downvoted.png'
 import upvoted from '../../assets/upvoted.png'
 
-const TruckCard = ({ truck, addFav }) => {
+const TruckCard = (props) => {
+  const [isFavorite, setIsFavorite] = useState(false)
+  const [isDownvoted, setIsDownvoted] = useState(false)
+  const [isUpvoted, setIsUpvoted] = useState(false)
+  console.log(props.truck, 'in truck card')
+
+const handleFav = () => {
+  setIsFavorite(!isFavorite)
+  if (!isFavorite) {
+    props.truck.favorited = true
+    props.addFav(props.truck)
+  } else {
+    props.truck.favorited = false
+    props.remFav(props.truck)
+  }
+}
+
+const handleUpvote = () => {
+  if (isDownvoted) {
+    return
```

Comment:
If I am reading this correctly, it's saying if (isDownvoted) is true, do nothing, meaning it's already selected.

If i am interpreting that right, maybe we could add a 
h3 that says "Already voted" or something like that. But looks great!